deploy/crds/devconsole_v1alpha1_gitsource_crd.yaml

```diff
@@ -58,18 +58,33 @@ spec:
                     to access the git repo
                   type: string
               required:
-              - name
+                - name
               type: object
             url:
               description: URL of the git repo
               type: string
           required:
-          - url
+            - url
           type: object
         status:
-          type: object
+          properties:
+            state:
+              description: State represents current state of the GitSource, can be either initializing or ready
+              type: string
+              minimum: 1
+            connection:
+              description: Connection holds information whether the last attempt to reach the git source
+              was successful or not. Optional
+              properties:
+                state:
+                  description: State is the result of the attempt to reach a GitSource. Can be either Failed or OK
+                  type: string
+                  mimimum: 1
+                error:
+                  description: Error has the error message if the attempt to reach a GitSource failed
```

Comment:
@christianvogt
From a UI/consumption perspective,
1. if there's `error` and the `advanced -> private repo` options were hidden, the 
 `advanced -> private repo` should show up.

2.  if there's `error` and the `advanced -> private repo` options were NOT hidden, they should continue showing up.

Reply:
Should we also capture a code to indicate whether the error is due to authentication, invalid url, unreachable, etc... ?
That way the UI can more clearly present the authentication section only when we detect it to be a private repository.

Reply:
We could consider including the `http` status code of the last attempt to reach the git url? Is that what you are suggesting @christianvogt ?
Or, are you looking for an enumerated list of error constants?

Reply:
```
curl --verbose https://github.com/redhWRONGoper/devopsconsole-s?service=git-upload-pack
HTTP/1.1 401 Authorization Required
```

```
curl --verbose https://bitbucket.org/sbose78/tedxbangalore2017.git/info/refs?service=git-upload-pack
HTTP/2 401 
```

So, for both an incorrect URL and a private URL, the same response status is returned even thought the body had friendly textual information. 

I would say, let's not parse the text in the body to map to custom error codes since their could be way too many possibilities there. The client/UI could use the http response status code to hide/unhide parts?

Reply:
If the status code is present, that would help.
401 we show the form to add auth.
404, or anything else, we simply show an error that the url is invalid.

Reply:
@christianvogt unfortunately, GH responds with the same status code for both cases:
* when a user doesn't have sufficient permissions (eg. private repo) 
* when a user provided a wrong URL or branch. 

So in the case of GH, we cannot differentiate the failures

Reply:
Ok. What about when the user enters the wrong domain altogether?

Reply:
@christianvogt I would be a bit careful with using the domain to identity anything because enterprise gitlab/github/bitbucket deployments have custom domains like source.ibm.com :)

Reply:
I was referring to status code when pinging a domain that doesn't exist at all. Not domain name inspection.